muikku-rest/src/main/java/fi/otavanopisto/muikku/rest/PermissionRESTService.java

```diff
@@ -30,24 +46,157 @@
+      return Response.status(Status.FORBIDDEN).build();
+    }
+    
+    List<UserGroupEntity> userGroupEntities;
+    
+    String permissionGroupIds = pluginSettingsController.getPluginSetting("workspace", "permission-group-ids");
+    if (permissionGroupIds == null) {
+      userGroupEntities = userGroupEntityController.listUserGroupEntities();
+    }
+    else {
+      userGroupEntities = new ArrayList<UserGroupEntity>();
+      String[] idArray = permissionGroupIds.split(",");
+      for (int i = 0; i < idArray.length; i++) {
+        Long groupId = NumberUtils.createLong(idArray[i]);
+
+        if (groupId != null) {
+          UserGroupEntity userGroupEntity = userGroupEntityController.findUserGroupEntityById(groupId);
+          if (userGroupEntity == null) {
+            logger.warning(String.format("Missing group %d in plugin setting workspace.permission-group-ids", groupId));
+          }
+          else {
+            userGroupEntities.add(userGroupEntity);
+          }
+        }
+        else {
+          logger.warning(String.format("Malformatted plugin setting workspace.permission-group-ids %s", permissionGroupIds));
+        }
+      }
+    }
+
+    List<Permission> permissions = listWorkspaceSettingsPermissions();
+    List<WorkspaceSettingsUserGroup> userGroupRestModels = new ArrayList<>();
+    
+    for (UserGroupEntity userGroupEntity : userGroupEntities) {
+      UserGroup userGroup = userGroupController.findUserGroup(userGroupEntity);
+
+      if (userGroup != null) {
+        WorkspaceSettingsUserGroup workspaceSettingsUserGroup = new WorkspaceSettingsUserGroup(workspaceEntityId, userGroupEntity.getId(), userGroup.getName());
+        
+        for (Permission permission : permissions) {
+          WorkspaceGroupPermission workspaceGroupPermission = permissionController.findWorkspaceGroupPermission(workspaceEntity, userGroupEntity, permission);
+          if (workspaceGroupPermission != null) {
+            workspaceSettingsUserGroup.addPermission(permission.getName());
+          }
+        }
+        
+        userGroupRestModels.add(workspaceSettingsUserGroup);
+      }
+    }
+    
+    userGroupRestModels.sort(Comparator.comparing(WorkspaceSettingsUserGroup::getUserGroupName));
+    
+    return Response.ok(userGroupRestModels).build();
+  }
+
+  @PUT
+  @Path ("/workspaceSettings/{WORKSPACEENTITYID}/userGroups/{USERGROUPID}")
+  @RESTPermit(handling = Handling.INLINE, requireLoggedIn = true)
+  public Response setWorkspaceSettingsUserGroup(@PathParam("WORKSPACEENTITYID") Long workspaceEntityId,
+      @PathParam("USERGROUPID") Long userGroupEntityId, WorkspaceSettingsUserGroup payload) {
+    if (!Objects.equals(workspaceEntityId, payload.getWorkspaceEntityId()) || !Objects.equals(userGroupEntityId, payload.getUserGroupEntityId())) {
+      return Response.status(Status.BAD_REQUEST).build();
+    }
+    
+    UserGroupEntity userGroupEntity = userGroupEntityController.findUserGroupEntityById(userGroupEntityId);
```

Comment:
This needs to be restricted to the groups in organizations the user has access to.